packages/fxa-settings/src/models/contexts/SettingsContext.ts

```diff
@@ -95,17 +95,29 @@ export function initializeSettingsContext() {
   const alertBarInfo = new AlertBarInfo();
   const apolloClient = createApolloClient(config.servers.gql.url);
 
+  const GET_UID_QUERY = gql`
+    query GetUid {
+      account {
+        uid
+      }
+    }
+  `;
+
   const isForCurrentUser = (event: Event) => {
-    const { account } = apolloClient.cache.readQuery<{ account: Account }>({
-      query: gql`
-        query GetUid {
-          account {
-            uid
-          }
-        }
-      `,
-    })!;
-    return account.uid === (event as CustomEvent).detail.uid;
+    let currentUid: string | undefined;
+
+    try {
+      const data = apolloClient.readQuery<{ account: { uid: string } }>({
+        query: GET_UID_QUERY,
+      });
+      currentUid = data?.account?.uid;
```

Comment:
Just curious, why is this one wrapped in a try catch but the other call, e.g. `this.apolloClient.readQuery<{ account: AccountData }>...` isn't?

Reply:
Good call, I don't think the try catch is strictly necessary here.